### ISSUE TITLE
Re-search for cover art on rename

### ISSUE THREAD

**Issue by orenwolf** [jasarien/Provenance#514]
Hey guys,

Thanks for all your hard work. I wish I had the iOS dev skillset to help on development, but unfortunately, all I can do is help test, report bugs, and suggest enhancements. To that end:

While I know cover art renaming is coming for tvOS, I was wondering if performing another search on rename was possible? In a lot of cases where the ROM checksum doesn't match, and the name is wrong, we end up cover-art-less. If a rename would allow for a search to succeed, then perhaps that's a simple feature addition that might make it possible for us to look up the box art ourselves and rename the ROM to match?

Thanks!

**Comment by sevdestruct** [comment]
@orenwolf: ok but what if i rename a file that i have custom art on (often the case with translated roms, especially as they don't match checksums and are common candidate for custom art, considering)?

I think the better way to go would be to consider the custom rename when a user manually requests a cover art to download on a specific rom, rather than every time a file is renamed..

**Comment by JoeMatt** [comment]
This is kind of moot, the search isn't based on the display name, it's based on the filename. When you rename a game in Provenance it only renames the 'title' entry in the database, the filename is the same as it is on import.

**Comment by sevdestruct** [comment]
I agree, @joe, only reason for this would be... (updating name of Issue) There. haha.

**Comment by JoeMatt** [comment]
This would be pretty easy to add, just add additional fallback searches to the part where we search for filename. 

But the question is, what if multiple ROMs match by the display title? You want to set them all?

**Comment by sevdestruct** [comment]
I think we should only do this when you long press on a single game, otherwise people are going to lose custom game art (that is unless/until #730 is implemented)

**Comment by JoeMatt** [comment]
I think I understand what this ticket is asking now,

1. A rom is imported correctly into a system but,
2. It didn't match the OpenPV database therefore,
3. The box art is the default blank.

You want,

1. Rename the 'title' field of the rom,
2. if it wasn't previously matched, meaning the PVGame.originalArtworkURL.isEmpty,
3. Search the database again for that system the game is already in , but by the new 'title' field.

I suppose this is moderately easy to do.

**Comment by sevdestruct** [comment]
Bingo.

**Comment by JoeMatt** [comment]
@claude this should't be too hard to do on it's own, basically if a database entry is incomplete, let's say the user imports roms with shitty filenames and a non-matched md5, we probably have a hard time making a meaningful useful dataabase search

then the user renames the game in the app to the proper name, this would be a good time to do a new metadata/artwork lookup for database entries that would benefit from that.

one small concern, i don't have to trigger re-searching blindly, like if a user has custom edited metadata, then renames it, and we stupidly rematch or mismatch it and overwrite all their edits, so this should be somewhat logical in what and when it should update metadata and artwork etc

also, we have some other expanded artwork and game manual scanning and import tickets here for planned updates, only lightly related but maybe worth noting.

make epic tickets for this research planning and work
